specs/ledger/hs/src/Ledger/Core.hs

```diff
@@ -83,17 +83,17 @@ verify (VKey vk) vd (Sig sd sk) = vk == sk && vd == sd
 -- Slots and Epochs
 ---------------------------------------------------------------------------------
 
-newtype Epoch = Epoch Natural
+newtype Epoch = Epoch Word64
```

Comment:
Is converting from `Natural` to `Word64` that annoying? I'd have thought there would be a preference for the more specific type here.

Reply:
It is annoying to me, but maybe there are better ways to circumvent it. At the executable spec level, we can choose to ignore the concrete domain of `Slot`s and `Epoch`s, however the problem comes when I want to elaborate these into the concrete type: what do I do if I get a number our of range?  I could not find a proper way of solving this when elaborating an abstract block into a concrete one, to feed to the chain validation function.

If there are ways of solving this, I'd prefer using `Natural`s, but restricting the (abstract) domain was the only way I could think of at the moment. Any suggestions/thoughts?